client/my-sites/post-type-list/post-actions-ellipsis-menu/copy-link.jsx

```diff
@@ -0,0 +1,64 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import PopoverMenuItemClipboard from 'components/popover/menu-item-clipboard';
+import { getPost } from 'state/posts/selectors';
+import { bumpStat, recordTracksEvent } from 'state/analytics/actions';
+import { bumpStatGenerator } from './utils';
+import { infoNotice } from 'state/notices/actions';
+
+function PostActionsEllipsisMenuCopyLink( { onCopyLinkClick, copyLink, infoNotice, translate } ) {
+	const onCopy = () => {
+		onCopyLinkClick();
+		infoNotice( translate( 'Link copied to clipboard.' ), { duration: 3000 } );
```

Comment:
I was originally just going to put this in `onCopyLinkClick`, but couldn't find a way to get `infoNotice` available in `mergeProps` (I thought it would be available at `dispatchProps.infoNotice`, but I kept getting errors). Lemme know if I've missed an easier way).

Reply:
Oh, this shows in the failed tests: `'infoNotice' is already declared in the upper scope.`.

Reply:
There is a conflict between the imported action and the name of the prop, so make sure the property received as part of the function arg is renamed to something else like `showInfoNotice`:

```js

import { infoNotice } from 'state/notices/actions';

function PostActionsEllipsisMenuCopyLink( { onCopyLinkClick, copyLink, showInfoNotice, translate } ) {
  showInfoNotice( ... );
}

const mapDispatchToProps = { bumpStat, showInfoNotice: infoNotice, recordTracksEvent };

Reply:
> I was originally just going to put this in onCopyLinkClick, but couldn't find a way to get infoNotice available in mergeProps (I thought it would be available at dispatchProps.infoNotice, but I kept getting errors). Lemme know if I've missed an easier way).

Maybe relate to the missing `type` prop I noted in https://github.com/Automattic/wp-calypso/pull/37049/files#r339102346 ?

The code above should work:

```js
// ...
import { translate } from 'i18n-calypso';
// ...
const PostActionsEllipsisMenuCopyLink = ( { onCopyLinkClick, copyLink } ) => (
	<PopoverMenuItemClipboard text={ copyLink } onCopy={ onCopyLinkClick }>
		{ translate( 'Copy Link' ) }
	</PopoverMenuItemClipboard>
);
// ...
const mapStateToProps = ( state, { globalId } ) => {
	// ...
	return {
		copyLink: post.URL,
		type: post.type,
	};
};
// ...
const mergeProps = ( stateProps, dispatchProps, ownProps ) => {
	// ...
	const onCopyLinkClick = () => {
		// ...
		dispatchProps.infoNotice( translate( 'Link copied to clipboard.' ), { duration: 3000 } );
	};
	// ...
};
// ...
```

Reply:
Ah, there we  go; fixing the missing type and properly making `translate` available got me what I was after. Thanks! 45b0b1df1a8f38ef8f543dc86f2bdb5507ef3fbe 837f62e3aa873fd9a796881d619dd394b2a00898